pygmt/tests/test_meca.py

```diff
@@ -0,0 +1,241 @@
+
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_meca_spec_1d_array():
+    """
+    Test supplying a 1D numpy array containing focal mechanisms and
+    locations to the `spec` argument.
+    """
+
+    fig = Figure()
+
+    # supply focal mechanisms to meca as a 1D numpy array, here we are using
+    # the Harvard CMT zero trace convention but the focal mechanism
+    # parameters may be specified any of the available conventions. Since we
+    # are not using a dict or dataframe the convention and component should
+    # be specified.
+    focal_mechanism = [
+        -127.40,  # longitude
+        40.87,  # latitude
+        12,  # depth
+        -3.19,  # mrr
+        0.16,  # mtt
+        3.03,  # mff
+        -1.02,  # mrt
+        -3.93,  # mrf
+        -0.02,  # mtf
+        23,  # exponent
+        0,  # plot_lon, 0 to plot at event location
+        0,  # plot_lat, 0 to plot at event location
+    ]
+    focal_mech_array = np.asarray(focal_mechanism)
+
+    fig.meca(
+        focal_mech_array,
+        convention="mt",
+        component="full",
+        region=[-128, -127, 40, 41],
+        scale="2c",
+        projection="M14c",
+    )
+
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_meca_spec_2d_array():
+    """
+    Test supplying a 2D numpy array containing focal mechanisms and
+    locations to the `spec` argument.
+    """
+
+    fig = Figure()
+
+    # supply focal mechanisms to meca as a 2D numpy array, here we are using
+    # the GCMT convention but the focal mechanism parameters may be
+    # specified any of the available conventions. Since we are not using a
+    # dict or dataframe the convention and component should be specified.
+    focal_mechanisms = [
+        [
+            -127.40,  # longitude
+            40.87,  # latitude
+            12,  # depth
+            170,  # strike1
+            20,  # dip1
+            -110,  # rake1
+            11,  # strike2
+            71,  # dip2
+            -83,  # rake2
+            5.1,  # mantissa
+            23,  # exponent
+            0,  # plot_lon, 0 means we want to plot at the event location
+            0,  # plot_lat
+        ],
+        [-127.50, 40.88, 12.0, 168, 40, -115, 20, 54, -70, 4.0, 23, 0, 0],
+    ]
+    focal_mechs_array = np.asarray(focal_mechanisms)
+
+    fig.meca(
+        focal_mechs_array,
+        convention="gcmt",
+        region=[-128, -127, 40, 41],
+        scale="2c",
+        projection="M14c",
+    )
+
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_meca_spec_file():
+    """
+    Test supplying a file containing focal mechanisms and locations to the
+    `spec` argument.
+    """
+
+    fig = Figure()
+
+    focal_mechanism = [-127.43, 40.81, 12, -3.19, 1.16, 3.93, -1.02, -3.93, -1.02, 23]
+
+    # writes temp file to pass to gmt
+    with open(os.path.join(TEST_DATA_DIR, "temp.test"), mode="w") as temp_file:
```

Comment:
I think this is a really good application for the GMTTempFile, docstring w/ usage example below:

https://github.com/GenericMappingTools/pygmt/blob/2884fd04a20ef6281c33d75de6c4521ba968b032/pygmt/helpers/tempfile.py#L27

Reply:
I think the usage example would be:
```python
from pygmt.helpers import GMTTempFile
with GMTTempFile() as temp:
    with open(temp.name, mode="w") as temp_file:
        temp_file.write(...)
    some_gmt_module(temp.name, ...)
```
Then you don't need to use the `os.remove()` since we have the context manager. LMK if that has issues!

Reply:
Oooo thanks Liam, that's clever! Testing this test is a little slow on my end because `make test` on my machine freezes on `../pygmt/tests/test_grdview.py::test_grdview_grid_dataarray` every time starting today -_-

Reply:
Finally passed after a restart but the grdview tests are very slow now. Looks like test_meca is passing now.

Reply:
Sorry about the slow test (we'll need to streamline this somehow at some point). A workaround is to only run the meca tests locally. Use `pytest -v --mpl pygmt/tests/test_meca.py`.

Reply:
The changes in 6c3af68c0364d2d163909fd648716b1bd26ca1be doesn't make use of `GMTTempFile`. The example given by @liamtoney is correct. In that example, the temporary file is written to the file defined by the variable `temp.name` and deleted when exiting GMTTempFile context. You should follow that example instead.

Reply:
Thanks @weiji14 that's very helpful.

Reply:
FYI, you can always use a command like below to run the tests in only one script:
```
pytest --mpl --mpl-results-path=results pygmt/tests/test_meca.py
```

Reply:
@seisman can you please explain? I used GMTTempFile in what I believe is the correct syntax. Including the `as temp` that Liam mentioned in the second line will throw a pylint warning for an unused variable. The relevant block is:
```    with GMTTempFile():
        with open("temp.name", mode="w") as temp_file:
            temp_file.write(" ".join([str(x) for x in focal_mechanism]))
        # supply focal mechanisms to meca as a file
        fig.meca(
            "temp.name",
            convention="mt",
            component="full",
            region=[-128, -127, 40, 41],
            scale="2c",
            projection="M14c",
        )```

Reply:
Your codes write the data to a file called `temp.name`, and the file is not deleted. 

You can think GMTTempFile as a wrapper of the Python [`tempfile.TemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile). 

A similar use case is: 
https://github.com/GenericMappingTools/pygmt/blob/4cfb7a2fdc165e8cdeeab88027b3fb09e0eaa134/pygmt/tests/test_text.py#L284-L297

Reply:
> Including the `as temp` that Liam mentioned in the second line will throw a pylint warning for an unused variable.

After using `with GMTTempFile() as temp:`, you should also remove the double quotes around `"temp.name"`.

Reply:
I see, so the temp.file object needed to be passed to meca rather than a temporary file name string. Should be fixed now.

Reply:
Yes, I think this PR is good to merge. Thanks for your contribution!

Reply:
`temp.name` is a string with the system filename.

Reply:
thank @liamtoney I apparently need some rest